ledger-tool/src/main.rs

```diff
@@ -741,6 +746,31 @@ fn open_blockstore(
     }
 }
 
+fn shred_storage_type_from_matches(
+    matches: &ArgMatches,
+    arg_rocksdb_shred_compaction: &str,
+    arg_rocksdb_fifo_shred_storage_size: &str,
+) -> ShredStorageType {
+    match matches.value_of(arg_rocksdb_shred_compaction) {
+        None => ShredStorageType::default(),
+        Some(shred_compaction_string) => match shred_compaction_string {
+            "level" => ShredStorageType::RocksLevel,
+            "fifo" => {
+                let shred_storage_size = match matches.value_of(arg_rocksdb_fifo_shred_storage_size)
+                {
+                    Some(_) => value_t_or_exit!(matches, arg_rocksdb_fifo_shred_storage_size, u64),
```

Comment:
@steviez: this one might be the blocker for the fully automatic determining the shred compaction, as fifo compaction requires a size.

Reply:
>as fifo compaction requires a size.

Most of the ledger-tool commands are read-only via Secondary mode. In these cases, we don't want to do any compactions as we don't want to alter any state. I'm not sure the API has a way to specify this, but using a very-large value (ie u64::MAX) would do the trick right to cover this right?

I think the only case where we actually care about this value is the `purge` command, as we may compact after that. So, we only need to check for a size for `purge` command when we are using `fifo`, right? Does that makes things workable in terms of auto detection?

Reply:
I think that would work, although I need to add a TODO as this might change the content inside the rocksdb files which makes us think u64::MAX was actually used.  Need to correct the u64::MAX workaround after the rocksdb options-file is used to detect the previous rocksdb options.